homeassistant/components/esphome/manager.py

```diff
@@ -268,11 +274,30 @@ def async_on_service_call(self, service: HomeassistantServiceCall) -> None:
         elif self.entry.options.get(
             CONF_ALLOW_SERVICE_CALLS, DEFAULT_ALLOW_SERVICE_CALLS
         ):
-            hass.async_create_task(
-                hass.services.async_call(
-                    domain, service_name, service_data, blocking=True
+            call_id = service.call_id
+            if call_id and service.wants_response:
+                # Service call with response expected
+                hass.async_create_task(
```

Comment:
nit: we should be using `config_entry.async_create_task` so they get cancelled automatically at tear down in case they are calling long running services

Reply:
bda8bcfd261cd949678ddaa2dd46a6a34e6129d5